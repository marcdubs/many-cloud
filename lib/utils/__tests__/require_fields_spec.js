const require_fields = require("../require_fields");

describe("require fields", () => {
  describe("when an undefined object is passed in", () => {
    it("throws an error", () => {
      expect(() => {
        require_fields();
      }).toThrowErrorMatchingSnapshot();
    });
  });

  describe("when fields is", () => {
    describe("undefined", () => {
      it("throws an error", () => {
        expect(() => {
          require_fields({});
        }).toThrowErrorMatchingSnapshot();
      });
    });

    describe("not an array", () => {
      it("throws an error", () => {
        expect(() => {
          require_fields({}, "not_an_array");
        }).toThrowErrorMatchingSnapshot();
      });
    });
  });

  describe("when an object does not contain a required field", () => {
    it("throws an error", () => {
      expect(() => {
        require_fields({}, ["a_key"]);
      }).toThrowErrorMatchingSnapshot();
    });
  });

  describe("when an object does contain a required field", () => {
    it("does not throw an error", () => {
      expect(() => {
        require_fields({ a_key: "hello" }, ["a_key"]);
      }).not.toThrowError();
    });
  });
});
