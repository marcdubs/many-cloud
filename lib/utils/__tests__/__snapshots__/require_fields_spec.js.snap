// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`require fields when an object does not contain a required field throws an error 1`] = `"Missing argument: \\"a_key\\"."`;

exports[`require fields when an undefined object is passed in throws an error 1`] = `"No object argument."`;

exports[`require fields when fields is not an array throws an error 1`] = `"fields argument isn't an array"`;

exports[`require fields when fields is undefined throws an error 1`] = `"No fields argument."`;
